### PR TITLE
Adding support for specific annotations for resources created

### PR DESCRIPTION
**Pull request by visheshtanksale**

### CODE REVIEW

File: api/apps/v1alpha1/common_types.go [hunk 1]
```diff
@@ -35,7 +35,8 @@ type Service struct {
 	// override the default service name
 	Name string `json:"name,omitempty"`
 	// +kubebuilder:default=8000
-	OpenAIPort int32 `json:"openaiPort"`
+	OpenAIPort         int32             `json:"openaiPort"`
+	ServiceAnnotations map[string]string `json:"serviceAnnotations,omitempty"`
```
Comment (shivamerla) [inline api/apps/v1alpha1/common_types.go:39]: This should be just `Annotations` as service.annotations is self explanatory and also it stutters.

File: api/apps/v1alpha1/common_types.go [hunk 2]
```diff
@@ -77,8 +79,9 @@ type Image struct {
 // Ingress defines attributes to enable ingress for the service
 type Ingress struct {
 	// ingress, or virtualService - not both
-	Enabled *bool                    `json:"enabled,omitempty"`
-	Spec    networkingv1.IngressSpec `json:"spec,omitempty"`
+	Enabled            *bool                    `json:"enabled,omitempty"`
+	IngressAnnotations map[string]string        `json:"ingressAnnotations,omitempty"`
```
Comment (shivamerla) [inline api/apps/v1alpha1/common_types.go:83]: same here

File: api/apps/v1alpha1/common_types.go [hunk 3]
```diff
@@ -47,9 +48,10 @@ type Metrics struct {
 
 // ServiceMonitor defines attributes to create a service monitor
 type ServiceMonitor struct {
-	AdditionalLabels map[string]string `json:"additionalLabels,omitempty"`
-	Interval         promv1.Duration   `json:"interval,omitempty"`
-	ScrapeTimeout    promv1.Duration   `json:"scrapeTimeout,omitempty"`
+	AdditionalLabels      map[string]string `json:"additionalLabels,omitempty"`
```
Comment (shivamerla) [inline api/apps/v1alpha1/common_types.go:51]: AdditionalLabels are not labels appended to the ServiceMonitor metadata, but instead they are additional labels added to the metrics collected from the target.

File: api/apps/v1alpha1/common_types.go [hunk 4]
```diff
@@ -47,9 +48,10 @@ type Metrics struct {
...
+	AdditionalAnnotations map[string]string `json:"additionalAnnotations,omitempty"`
```
Comment (shivamerla) [inline api/apps/v1alpha1/common_types.go:52]: same here, just "Annotations" should be used.

File: api/apps/v1alpha1/common_types.go (hunk 1 above)
Comment (visheshtanksale) [reply api/apps/v1alpha1/common_types.go]: Fixed

File: api/apps/v1alpha1/common_types.go (hunk 2 above)
Comment (visheshtanksale) [reply api/apps/v1alpha1/common_types.go]: Fixed

File: api/apps/v1alpha1/common_types.go (hunk 3 above)
Comment (visheshtanksale) [reply api/apps/v1alpha1/common_types.go]: Noted

File: api/apps/v1alpha1/common_types.go (hunk 4 above)
Comment (visheshtanksale) [reply api/apps/v1alpha1/common_types.go]: Fixed

**Review by shivamerla — APPROVED** [review]
LGTM

**Pull request merged**